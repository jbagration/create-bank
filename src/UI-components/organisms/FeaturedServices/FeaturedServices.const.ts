import { FC } from 'react'
import {
    BankingServicesIcon,
    InternetIcon,
    MobileIcon,
    OtherIcon,
    SberIcon,
    TransferEasyIcon,
    TransferIcon,
    UtilitiesIcon,
} from 'src/assets'

export const featuredInfo = [
    'Пополнение телефона',
    'Перевод на карту',
    'Коммунальные услуги',
    'Обмен валюты',
    'Добавить',
]

export type LinkMapping = Record<string, string>

const linkMapping: LinkMapping = {
    'Пополнение телефона': 'user_main/payments/infopage/mobile',
    'Коммунальные услуги': 'user_main/payments/utilities',
    'Обмен валюты': 'exchange_rate',
}
export interface DefaultPayment {
    text: string
    Icon: FC
    href: string
    data: null | {}
}
export const popularDefaultPayments: DefaultPayment[] = [
    {
        text: 'Банковские услуги',
        Icon: BankingServicesIcon,
        href: 'payments/financepage',
        data: null,
    },
    {
        text: 'Коммунальные услуги',
        Icon: UtilitiesIcon,
        href: 'payments/utilities',
        data: null,
    },
    {
        text: 'Мобильная связь',
        Icon: MobileIcon,
        href: 'payments/infopage/mobile',
        data: null,
    },
    {
        text: 'Интернет',
        Icon: InternetIcon,
        href: 'payments/infopage/internet',
        data: null,
    },
    {
        text: 'Перевод на карту',
        Icon: TransferIcon,
        href: 'transfers',
        data: null,
    },
    {
        text: 'Прочие услуги',
        Icon: OtherIcon,
        href: 'payments/other',
        data: null,
    },
]

interface PopularPayments {
    transferMySelf: DefaultPayment
    transferToAnotherCard: DefaultPayment
}
export const popularPayments: PopularPayments = {
    transferMySelf: {
        href: 'transfers/myself',
        text: 'Перевод между своими счетами',
        Icon: TransferEasyIcon,
        data: null,
    },
    transferToAnotherCard: {
        href: 'transfers/card',
        text: 'Перевод на карту другого банка',
        Icon: SberIcon,
        data: null,
    },
}
export default linkMapping
