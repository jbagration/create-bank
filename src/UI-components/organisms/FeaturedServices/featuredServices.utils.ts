import { OrderItem, TransferOrdersItem } from 'src/UI-components/molecules'

import {
    popularDefaultPayments,
    popularPayments,
} from './FeaturedServices.const'

const last = 0

export function getPopularPayments(
    operations: TransferOrdersItem[] | undefined
) {
    const listIsEmptyOrUndefined = !operations || !operations.length

    if (listIsEmptyOrUndefined) {
        return popularDefaultPayments
    }
    const lastDate = operations && operations[last].date
    const lastOperations = filterOperationsByDate(operations, lastDate)
    const payments = getPayments(lastOperations)

    return getPopular(payments)
}

function filterOperationsByDate(
    payments: TransferOrdersItem[],
    d: string
): TransferOrdersItem[] {
    const date = getMonthAndDay(d)
    return payments.filter((payment) => {
        const currentDate = getMonthAndDay(payment.date)
        return currentDate === date
    })
}

function getPayments(payments: TransferOrdersItem[]): OrderItem[] {
    return payments.reduce<OrderItem[]>((acc, current) => {
        return [...acc, ...current.orders]
    }, [])
}

function getPopular(payments: OrderItem[]) {
    const listIsEmpty = payments.length < 1
    if (listIsEmpty) {
        return popularDefaultPayments
    }

    const listOfPaymentsType = Array.from(
        new Set(payments.map((payment) => payment.transferTypeName))
    )

    const listByPopularity = listOfPaymentsType
        .map((el) =>
            payments.filter((payment) => payment.transferTypeName === el)
        )
        .sort((first, next) => (first.length > next.length ? -1 : 1))
        .map((payments) => {
            if (payments[last].transferTypeName === 'BETWEEN_CARDS') {
                return popularPayments.transferMySelf
            }

            if (payments[last].transferTypeName === 'TO_ANOTHER_CARD') {
                return popularPayments.transferToAnotherCard
            }

            return popularPayments.transferToAnotherCard
        })
    return [...listByPopularity, ...popularDefaultPayments].slice(0, 6)
}

function getMonthAndDay(date: string) {
    const regExp = /\.(.*)$/
    return (date.match(regExp) as RegExpMatchArray)[0]
}
