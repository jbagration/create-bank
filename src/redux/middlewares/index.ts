export { errorNotificationMiddleware } from './errorNotification'
export { logoutMiddleware } from './logout'
