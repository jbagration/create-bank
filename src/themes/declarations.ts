export {}

declare module '@mui/material/TextField' {
    interface TextFieldPropsSizeOverrides {
        short: true
        long: true
        medium: false
        small: false
    }
}
declare module '@mui/material/InputBase' {
    export interface InputBaseComponentsPropsOverrides {
        isFocused?: boolean
    }
}
