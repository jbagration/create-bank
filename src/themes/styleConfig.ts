import { createTheme } from '@mui/material/styles'

import RobotoRegular from '../assets/fonts/Roboto-Regular.woff2'
import RobotoBlack from '../assets/fonts/Robotoblack.woff2'
import RobotoBold from '../assets/fonts/Robotobold.woff2'
import RobotoMedium from '../assets/fonts/Robotomedium.woff2'

import {
    blackColor,
    darkGrayColor,
    grayColor,
    systemColor,
    whiteColor,
    yellowColor,
} from './colors/_index'
import { specificTextFont } from './typography/specificText'
import {
    bodyFont,
    buttonFont,
    headlineFont,
    strongFont,
    subtitleFont,
    systemFont,
    tagFont,
    textLinkFont,
} from './typography'

export const Breakpoints = {
    xs: 0,
    sm: 600,
    md: 960,
    lg: 1280,
    lp: 1440,
    xl: 1920,
}

export const lightTheme = createTheme({
    typography: () => ({
        fontSize: 16,
        lineHeight: '20px',
        fontWeight: 400,
        fontFamily: ['Roboto', 'Montserrat', 'Open Sans'].join(','),
        ...buttonFont,
        ...headlineFont,
        ...subtitleFont,
        ...strongFont,
        ...bodyFont,
        ...tagFont,
        ...textLinkFont,
        ...systemFont,
        ...specificTextFont,
    }),

    components: {
        MuiCssBaseline: {
            styleOverrides: `  @font-face {
          font-family: 'Roboto';
          src: url('${RobotoRegular}') format('woff2');
          font-weight: 400;
          font-style: normal;
        };
         @font-face {
          font-family: 'Roboto';
          src: url('${RobotoMedium}') format('woff2');
          font-weight: 500;
          font-style: normal;
        };
         @font-face {
          font-family: 'Roboto';
          src: url('${RobotoBold}') format('woff2');
          font-weight: 700;
          font-style: normal;
        };
         @font-face {
          font-family: 'Roboto';
          src: url('${RobotoBlack}') format('woff2');
          font-weight: 900;
          font-style: normal;
        };
      ,
`,
        },

        MuiTextField: {
            styleOverrides: {
                root: {
                    '& label': {
                        color: grayColor['900'],
                    },
                },
            },
        },

        MuiAppBar: {
            styleOverrides: {
                colorPrimary: {
                    backgroundColor: 'transparent',
                    boxShadow: 'none',
                },
            },
        },

        MuiButton: {
            styleOverrides: {
                contained: {},
                outlined: {},
                text: {},
            },
        },
    },

    spacing: 4,
    breakpoints: {
        values: {
            lg: Breakpoints.lg,
            md: Breakpoints.md,
            sm: Breakpoints.sm,
            xl: Breakpoints.xl,
            xs: Breakpoints.xs,
        },
    },

    palette: {
        text: {
            primary: grayColor['900'],
        },
        Text: {
            TextPrimary: grayColor['900'],
            TextSecondary: whiteColor['0'],
            TextTertiary: grayColor['700'],
            TextQuinary: grayColor['700'],
            TextQuaternary: grayColor['900'],
            TextSenary: whiteColor['0'],
            TextDisabled: grayColor['500'],
            SemiText: grayColor['400'],
        },
        BackAccent: {
            BackgroundPrimary: yellowColor['700'],
            BackgroundHover: yellowColor.Hover,
            BackgroundBorder: yellowColor.Border,
            BackgroundSecondary: yellowColor['300'],
            BackgroundTertiary: yellowColor['50'],
            BackgroundQuaternary: yellowColor['300'],
            BackgoundFilter: yellowColor['100'],
        },
        BackGray: {
            BackgroundPrimary: grayColor['300'],
            BackgroundHover: grayColor['400'],
            BackgroundSecondary: grayColor['500'],
            BackgroundSlider: grayColor['200'],
            BackgoundFilter: grayColor['50'],
            BackgoundFilterHover: grayColor['100'],
            BackgoundDark: grayColor['900'],
        },
        BackStandart: {
            BackgroundPrimary: whiteColor[0],
            BackgroundSecondary: whiteColor[0],
            BackgroundTertiary: whiteColor[0],
            BackgroundQuaternary: grayColor['900'],
            BackgroundQuinary: whiteColor[0],
            BackgroundSenary: grayColor['200'],
            BackgroundSeptenary: grayColor['100'],
            BackgroundOctonary: grayColor['100'],
            BackgroundNonary: grayColor['300'],
        },
        SystemColor: {
            Success: systemColor.Success,
            Notification: systemColor.Notification,
            Error: systemColor.Error,
            Warning: systemColor.Warning,
        },
        BackgroundOpacity: {
            backgroundColor: blackColor[0],
        },
    },
})

export const darkTheme = createTheme({
    typography: () => ({
        fontSize: 16,
        lineHeight: '20px',
        fontWeight: 400,
        fontFamily: ['Roboto', 'Montserrat', 'Open Sans'].join(','),
        ...buttonFont,
        ...headlineFont,
        ...subtitleFont,
        ...strongFont,
        ...bodyFont,
        ...tagFont,
        ...textLinkFont,
        ...systemFont,
        ...specificTextFont,
    }),

    components: {
        MuiCssBaseline: {
            styleOverrides: `  @font-face {
          font-family: 'Roboto';
          src: url('${RobotoRegular}') format('woff2');
          font-weight: 400;
          font-style: normal;
        };
         @font-face {
          font-family: 'Roboto';
          src: url('${RobotoMedium}') format('woff2');
          font-weight: 500;
          font-style: normal;
        };
         @font-face {
          font-family: 'Roboto';
          src: url('${RobotoBold}') format('woff2');
          font-weight: 700;
          font-style: normal;
        };
         @font-face {
          font-family: 'Roboto';
          src: url('${RobotoBlack}') format('woff2');
          font-weight: 900;
          font-style: normal;
        };
      ,
    `,
        },

        MuiTextField: {
            styleOverrides: {
                root: {
                    '& label': {
                        color: grayColor['900'],
                    },
                },
            },
        },

        MuiAppBar: {
            styleOverrides: {
                colorPrimary: {
                    backgroundColor: 'transparent',
                    boxShadow: 'none',
                },
            },
        },

        MuiButton: {
            styleOverrides: {
                contained: {},
                outlined: {},
                text: {},
            },
        },
    },

    spacing: 4,
    breakpoints: {
        values: {
            lg: Breakpoints.lg,
            md: Breakpoints.md,
            sm: Breakpoints.sm,
            xl: Breakpoints.xl,
            xs: Breakpoints.xs,
        },
    },

    palette: {
        background: {
            default: darkGrayColor[500],
        },
        text: {
            primary: whiteColor[0],
        },
        Text: {
            TextPrimary: whiteColor[0],
            TextSecondary: whiteColor[0],
            TextTertiary: grayColor['500'],
            TextQuaternary: grayColor['900'],
            TextQuinary: grayColor['400'],
            TextSenary: grayColor['900'],
            TextDisabled: grayColor['400'],
            SemiText: whiteColor[0],
        },
        BackAccent: {
            BackgroundPrimary: yellowColor['700'],
            BackgroundHover: yellowColor.Hover,
            BackgroundBorder: yellowColor.Border,
            BackgroundSecondary: yellowColor['300'],
            BackgroundTertiary: darkGrayColor['200'],
            BackgroundQuaternary: darkGrayColor['200'],
            BackgoundFilter: darkGrayColor['100'],
        },
        BackGray: {
            BackgroundPrimary: grayColor['300'],
            BackgroundHover: grayColor['400'],
            BackgroundSecondary: grayColor['500'],
            BackgroundSlider: grayColor['700'],
            BackgoundFilter: darkGrayColor['400'],
            BackgoundFilterHover: grayColor['500'],
            BackgoundDark: grayColor['900'],
        },
        BackStandart: {
            BackgroundPrimary: grayColor['900'],
            BackgroundSecondary: darkGrayColor['300'],
            BackgroundTertiary: darkGrayColor['500'],
            BackgroundQuaternary: whiteColor[0],
            BackgroundQuinary: darkGrayColor['500'],
            BackgroundSenary: darkGrayColor['300'],
            BackgroundSeptenary: darkGrayColor['200'],
            BackgroundOctonary: grayColor['900'],
            BackgroundNonary: darkGrayColor['300'],
        },
        SystemColor: {
            Success: systemColor.Success,
            Notification: systemColor.Notification,
            Error: systemColor.Error,
            Warning: systemColor.Warning,
        },
        BackgroundOpacity: {
            backgroundColor: blackColor[0],
        },
    },
})

declare module '@mui/material/styles' {
    export interface ThemeOptions {
        spacing: number
    }
}
