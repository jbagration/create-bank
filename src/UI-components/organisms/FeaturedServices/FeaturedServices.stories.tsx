import { ComponentMeta, ComponentStory } from '@storybook/react'

import { FeaturedServices } from './FeaturedServices'

export default {
    title: 'UI-components/organisms/FeaturedServices',
    component: FeaturedServices,
    parameters: {
        design: {
            type: 'figspec',
            url: 'https://www.figma.com/file/Z8Y02skRyeUmJLVhym1Fwi/Polybank_web_new?node-id=5931%3A60309&t=6O69MFC0DW9gMAjh-4',
        },
    },
} as ComponentMeta<typeof FeaturedServices>

const Template: ComponentStory<typeof FeaturedServices> = () => (
    <FeaturedServices />
)

export const Default = Template.bind({})
