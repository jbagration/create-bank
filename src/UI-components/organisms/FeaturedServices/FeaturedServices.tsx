import { FC } from 'react'
import { Link, useNavigate } from 'react-router-dom'
import { cardsApi } from 'src/redux/api/cardsApi'
import { useGetOperationsHistoryQuery } from 'src/redux/api/historyApi'
import { useAppDispatch } from 'src/redux/hooks/redux'
import { Grid } from '@mui/material'

import { FeaturedItem } from '../../molecules/FeaturedItem/FeaturedItem'

import { getPopularPayments } from './featuredServices.utils'

export const FeaturedServices: FC = () => {
    const { data = [] } = useGetOperationsHistoryQuery({
        pageNumber: 0,
        pageSize: 100,
    })
    const dispatch = useAppDispatch()
    const navigate = useNavigate()

    const popularPayments = getPopularPayments(data)

    const handleClick = (href: string) => {
        dispatch(cardsApi.endpoints.getDepositCards.initiate()).then(
            (response) => {
                const { data: cards } = response
                navigate(href, { state: { cards } })
            }
        )
    }
    return (
        <Grid data-testid="featured" container>
            {popularPayments.map(({ href, Icon, text }) => (
                <Grid
                    key={text}
                    sx={{
                        display: 'flex',
                        alignItems: 'center',
                        flexDirection: 'column',
                    }}
                    item
                    md={2}
                >
                    <Link
                        onClick={() => handleClick(href)}
                        to="#"
                        style={{
                            textDecoration: 'none',
                            color: 'inherit',
                            display: 'flex',
                            alignItems: 'center',
                            flexDirection: 'column',
                        }}
                    >
                        <FeaturedItem operation={text} icon={<Icon />} />
                    </Link>
                </Grid>
            ))}
        </Grid>
    )
}
