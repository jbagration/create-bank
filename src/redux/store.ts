import {
    FLUSH,
    PAUSE,
    PERSIST,
    persistReducer,
    persistStore,
    PURGE,
    REGISTER,
    REHYDRATE,
} from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import { combineReducers, configureStore } from '@reduxjs/toolkit'

import { api } from './api/api'
import { geocoderApi } from './api/geocoderApi'
import { authReducer } from './slices/auth'
import { branchReducer } from './slices/bankBranchSlice'
import { creditProductFormReducer } from './slices/creditProductsSlice'
import { depositProductsReducer } from './slices/depositProductsSlice'
import { faqReducer } from './slices/faqSlice'
import { myCardsReducer } from './slices/myCardsSlice'
import { errorNotificationMiddleware, logoutMiddleware } from './middlewares'

const persistConfig = {
    key: 'root',
    version: 1,
    storage,
    blacklist: [
        api.reducerPath,
        geocoderApi.reducerPath,
        'branches',
        'cardProducts',
    ],
}

export const rootReducer = combineReducers({
    auth: authReducer,
    branches: branchReducer,
    depositProducts: depositProductsReducer,
    creditProductForm: creditProductFormReducer,
    faq: faqReducer,
    myCards: myCardsReducer,
    [geocoderApi.reducerPath]: geocoderApi.reducer,
    [api.reducerPath]: api.reducer,
})

const persistedReducer = persistReducer(persistConfig, rootReducer)

export const store = configureStore({
    reducer: persistedReducer,
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
            serializableCheck: {
                ignoredActions: [
                    FLUSH,
                    REHYDRATE,
                    PAUSE,
                    PERSIST,
                    PURGE,
                    REGISTER,
                ],
                ignoredPaths: [
                    'creditProductForm.passport',
                    'creditProductForm.incomeStatement',
                ],
            },
        }).concat(
            api.middleware,
            geocoderApi.middleware,
            logoutMiddleware,
            errorNotificationMiddleware
        ),
})

export const persistor = persistStore(store)
export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
