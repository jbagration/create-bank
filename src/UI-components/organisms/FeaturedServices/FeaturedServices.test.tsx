import { Provider } from 'react-redux'
import { ThemeProvider } from '@mui/material'
import { render, screen } from '@testing-library/react'

import { store } from '../../../redux/store'
import { lightTheme } from '../../../themes/styleConfig'

import { FeaturedServices } from './FeaturedServices'

test('snapshot featured services', () => {
    render(
        <Provider store={store}>
            <ThemeProvider theme={lightTheme}>
                <FeaturedServices />
            </ThemeProvider>
        </Provider>
    )

    expect(screen.getByTestId('featured')).toMatchSnapshot()
})
