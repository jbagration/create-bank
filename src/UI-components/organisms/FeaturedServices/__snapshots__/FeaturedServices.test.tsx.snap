// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot featured services 1`] = `
<div
  class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  data-testid="featured"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-2 css-fbfhad-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-12lx19j"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-p2kfw9-MuiSvgIcon-root"
        data-testid="PhoneIphoneIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 1h-8C6.12 1 5 2.12 5 3.5v17C5 21.88 6.12 23 7.5 23h8c1.38 0 2.5-1.12 2.5-2.5v-17C18 2.12 16.88 1 15.5 1zm-4 21c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm4.5-4H7V4h9v14z"
        />
      </svg>
    </div>
    <span
      class="MuiTypography-root MuiTypography-textLink1 css-1j4biek-MuiTypography-root"
    >
      Пополнение телефона
    </span>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-2 css-fbfhad-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-12lx19j"
    >
      <svg>
        Cards.svg
      </svg>
    </div>
    <span
      class="MuiTypography-root MuiTypography-textLink1 css-1j4biek-MuiTypography-root"
    >
      Перевод на карту
    </span>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-2 css-fbfhad-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-12lx19j"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-p2kfw9-MuiSvgIcon-root"
        data-testid="HomeWorkOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17 15h2v2h-2zm0-4h2v2h-2zm0-4h2v2h-2zm-3.26 0 1.26.84V7z"
        />
        <path
          d="M10 3v1.51l2 1.33V5h9v14h-4v2h6V3z"
        />
        <path
          d="M8.17 5.7 15 10.25V21H1V10.48L8.17 5.7zM10 19h3v-7.84L8.17 8.09 3 11.38V19h3v-6h4v6z"
        />
      </svg>
    </div>
    <span
      class="MuiTypography-root MuiTypography-textLink1 css-1j4biek-MuiTypography-root"
    >
      Коммунальные услуги
    </span>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-2 css-fbfhad-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-12lx19j"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-p2kfw9-MuiSvgIcon-root"
        data-testid="CurrencyExchangeOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12.89 11.1c-1.78-.59-2.64-.96-2.64-1.9 0-1.02 1.11-1.39 1.81-1.39 1.31 0 1.79.99 1.9 1.34l1.58-.67c-.15-.45-.82-1.92-2.54-2.24V5h-2v1.26c-2.48.56-2.49 2.86-2.49 2.96 0 2.27 2.25 2.91 3.35 3.31 1.58.56 2.28 1.07 2.28 2.03 0 1.13-1.05 1.61-1.98 1.61-1.82 0-2.34-1.87-2.4-2.09l-1.66.67c.63 2.19 2.28 2.78 2.9 2.96V19h2v-1.24c.4-.09 2.9-.59 2.9-3.22 0-1.39-.61-2.61-3.01-3.44zM3 21H1v-6h6v2H4.52c1.61 2.41 4.36 4 7.48 4 4.97 0 9-4.03 9-9h2c0 6.08-4.92 11-11 11-3.72 0-7.01-1.85-9-4.67V21zm-2-9C1 5.92 5.92 1 12 1c3.72 0 7.01 1.85 9 4.67V3h2v6h-6V7h2.48C17.87 4.59 15.12 3 12 3c-4.97 0-9 4.03-9 9H1z"
        />
      </svg>
    </div>
    <span
      class="MuiTypography-root MuiTypography-textLink1 css-1j4biek-MuiTypography-root"
    >
      Обмен валюты
    </span>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-2 css-fbfhad-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-12lx19j"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-p2kfw9-MuiSvgIcon-root"
        data-testid="AddOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </div>
    <span
      class="MuiTypography-root MuiTypography-textLink1 css-1j4biek-MuiTypography-root"
    >
      Добавить
    </span>
  </div>
</div>
`;
